crates/wasm-pkg-client/src/warg/mod.rs

```diff
@@ -35,13 +35,23 @@ impl WargBackend {
         let warg_meta = registry_meta
             .protocol_config::<WargRegistryMetadata>("warg")?
             .unwrap_or_default();
-        let url = warg_meta.url.unwrap_or_else(|| registry.to_string());
+
         let WargRegistryConfig {
             client_config,
             auth_token,
             ..
         } = registry_config.try_into()?;
 
+        let url = warg_meta.url.unwrap_or_else(|| {
+            // If we just pass registry plain, warg will assume it is https. This is a workaround to
+            // assume that a local registry is http.
+            if registry.host() == "localhost" || registry.host() == "127.0.0.1" {
```

Comment:
add `"0.0.0.0"`?

Reply:
0.0.0.0 is a weird one because it means all interfaces for some things and I think it is rejected (maybe?) on Windows